vllm/model_executor/custom_op.py

```diff
@@ -138,15 +156,29 @@ def default_on() -> bool:
     # - MyOp.enabled()
     # - op_registry["my_op"].enabled()
     op_registry: dict[str, type['CustomOp']] = {}
+    op_registry_oot: dict[str, type['CustomOp']] = {}
```

Comment:
Why is the registry separate for out of tree?

For context, PyTorch has a custom operator mechanism but the whole registry is shared for built-in and custom operators.

Reply:
I tried to use 'op_registry' previously, but because custom_op might be used by chained derived class, for example, RotaryEmbedding and its sub classes. 

Using 'op_regitry' to replace in-tree class with out-of-tree class, will cause name confliction. 
For example:
We only want to replace Llama4RotaryEmbedding to HPULlama4RotaryEmbedding, while Llama4RotaryEmbedding's custom_op name is 'rotary_embedding', same as other RotaryEmbedding classes. So we can't actually not only replaced Llama4RotaryEmbedding but we replaced RotaryEmbedding to use HPULlama4RotaryEmbedding.

To reuse custom_op mechanism without breaking its existing functionality, I think it will make the modification much clearer by introducing 'op_registry_oot' to only serve for class replacement for any oot class.